plugins/org.polarsys.capella.transition.system2subsystem/src/org/polarsys/capella/transition/system2subsystem/rules/fa/FunctionalChainInvolvementRule.java

```diff
@@ -253,21 +198,17 @@ private boolean isDirectJump(FunctionalChainInvolvement nextSrcValid, Functional
   }
 
   private Collection<FunctionalChainInvolvement> listInvolvments(FunctionalChainInvolvement startFci,
-      FunctionalChainInvolvement endFci) {
-    Collection<FunctionalChainInvolvement> res = new ArrayList<>();
-    FunctionalChainInvolvement current = startFci;
-
-    while (!current.getNextFunctionalChainInvolvements().isEmpty()) {
-      FunctionalChainInvolvement next = current.getNextFunctionalChainInvolvements().get(0);
-      if (next.equals(endFci) == false) {
-        current = next;
-        res.add(current);
-      } else {
-        break;
-      }
+      FunctionalChainInvolvement endFci, IContext context) {
```

Comment:
here, as before, we return the first path toward the endFci. We can add all of them if needed at some point.